### PR TITLE
Add core APIs to client library

### PR DESCRIPTION
**Pull request by mdegat01**
# Proposed Changes

Add support for Supervisor's Home Assistant Core APIs to client library

### CODE REVIEW

File: aiohasupervisor/models/homeassistant.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    version: str | None = None
+    backup: bool | None = None
+
+
+@dataclass(frozen=True, slots=True)
+class HomeAssistantRestartOptions(Options):
+    """HomeAssistantRestartOptions model."""
+
+    safe_mode: bool | None = None
+    force: bool | None = None
```
Comment (MartinHjelmare) [inline aiohasupervisor/models/homeassistant.py:64]: The `force` option isn't documented here that I can see.

https://developers.home-assistant.io/docs/api/supervisor/endpoints#core

File: aiohasupervisor/homeassistant.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    async def start(self) -> None:
+        """Start Home Assistant."""
+        await self._client.post("core/start")
+
+    async def check_config(self) -> None:
+        """Check Home Assistant config."""
+        await self._client.post("core/check")
+
+    async def rebuild(self, options: HomeAssistantRestartOptions | None = None) -> None:
```
Comment (MartinHjelmare) [inline aiohasupervisor/homeassistant.py:57]: Should we use the restart options here for rebuild? The options are the same at the moment so technically it works, but maybe they will diverge in the future?

File: aiohasupervisor/models/homeassistant.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+@dataclass(frozen=True, slots=True)
+class HomeAssistantStopOptions(Request):
+    """HomeAssistantStopOptions model."""
+
+    force: bool
```
Comment (MartinHjelmare) [inline aiohasupervisor/models/homeassistant.py:71]: This isn't documented either.

File: tests/test_homeassistant.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        ("POST", URL(f"{SUPERVISOR_URL}/core/options"))
+    }
+
+
+async def test_homeassistant_update(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test Home Assistant update API."""
+    responses.post(f"{SUPERVISOR_URL}/core/update", status=200)
+    assert await supervisor_client.homeassistant.update() is None
```
Comment (MartinHjelmare) [inline tests/test_homeassistant.py:72]: Should we test passing options too?

File: tests/test_homeassistant.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    assert responses.requests.keys() == {("POST", URL(f"{SUPERVISOR_URL}/core/update"))}
+
+
+async def test_homeassistant_restart(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test Home Assistant restart API."""
+    responses.post(f"{SUPERVISOR_URL}/core/restart", status=200)
+    assert await supervisor_client.homeassistant.restart() is None
```
Comment (MartinHjelmare) [inline tests/test_homeassistant.py:81]: Test passing options?

File: tests/test_homeassistant.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        ("POST", URL(f"{SUPERVISOR_URL}/core/restart"))
+    }
+
+
+async def test_homeassistant_stop(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test Home Assistant stop API."""
+    responses.post(f"{SUPERVISOR_URL}/core/stop", status=200)
+    assert await supervisor_client.homeassistant.stop() is None
```
Comment (MartinHjelmare) [inline tests/test_homeassistant.py:92]: Test passing options?

File: tests/test_homeassistant.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    assert await supervisor_client.homeassistant.check_config() is None
+    assert responses.requests.keys() == {("POST", URL(f"{SUPERVISOR_URL}/core/check"))}
+
+
+async def test_homeassistant_rebuild(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test Home Assistant rebuild API."""
+    responses.post(f"{SUPERVISOR_URL}/core/rebuild", status=200)
+    assert await supervisor_client.homeassistant.rebuild() is None
```
Comment (MartinHjelmare) [inline tests/test_homeassistant.py:119]: Test passing options?

File: aiohasupervisor/models/homeassistant.py (hunk 1 above)
Comment (mdegat01) [reply aiohasupervisor/models/homeassistant.py:64]: Oh I guess the PR for that kind of got lost in the shuffle. It's still open here: https://github.com/home-assistant/developers.home-assistant/pull/2260

File: aiohasupervisor/models/homeassistant.py (hunk 3 above)
Comment (mdegat01) [reply aiohasupervisor/models/homeassistant.py:71]: Yea same PR listed above is pending documenting the `force` option for all of these

File: aiohasupervisor/homeassistant.py (hunk 2 above)
Comment (mdegat01) [reply aiohasupervisor/homeassistant.py:57]: Yea that's fair. Its unlikely but safer to do that else we have to accept an object of either type in the future if they do diverge.

File: aiohasupervisor/homeassistant.py [hunk 8]
```diff
@@ -0,0 +1,61 @@
+"""Home Assistant client for supervisor."""
+
+from .client import _SupervisorComponentClient
+from .models.homeassistant import (
+    HomeAssistantInfo,
+    HomeAssistantOptions,
```
Comment (MartinHjelmare) [inline aiohasupervisor/homeassistant.py:6]:
```suggestion
    HomeAssistantOptions,
    HomeAssistantRebuildOptions,
```

File: aiohasupervisor/homeassistant.py (hunk 2 above)
Comment (MartinHjelmare) [inline aiohasupervisor/homeassistant.py:57]:
```suggestion
    async def rebuild(self, options: HomeAssistantRebuildOptions | None = None) -> None:
```

**Review by MartinHjelmare — APPROVED** [review]

File: aiohasupervisor/homeassistant.py (hunk 2 above)
Comment (mdegat01) [reply aiohasupervisor/homeassistant.py:57]: Shoot, thanks

**Pull request merged**